### ISSUE TITLE
QUEST - reading and editing the Octane Guides preview

### ISSUE THREAD

**Issue by jenweber** [ember-learn/guides-source#588]
Hello! Thanks in advance for your help! While Octane isn't officially released, and won't be for a while, we are providing a "preview" period where adventurous devs can try it out, and they need guides documentation! A lot of work has already been done, so at this stage, we are looking for people to read the draft of the Octane guides and help us fix errors and fill in the blanks. Can you help us polish things up? Word choice doesn't matter much, so if you see stylistic improvements, please open an issue instead of including them in a PR. However spelling, typo fixes, and and factual accuracy are very important and we would love your help with those.

All work should be done by branching from `octane` in this repository, and all PRs should be pointed at `octane`, not `master`. See https://github.com/ember-learn/guides-source/issues/583 for ideas of what to look for, and links to a script that will aggregate code blocks, if that is useful to you. Thank you!!!

If you can help, please comment on this issue to say which page within the guides you can review. In your comment, post a link. When you have a PR, in the description please link to this issue.

## List of pages that may need volunteers
(read comments below to see if they are taken)

- [x] "Getting Started", url: 'getting-started'
    - [x] "Quick Start", url: "quick-start", #590
    - [x] "Installing Ember", url: "index", checked during https://github.com/ember-learn/guides-source/pull/590
    - [x] "Core Concepts", url: "core-concepts" - needs an updated image
- [x] "Anatomy of an App", url: "anatomy-of-an-app"
    - [x] "Overview", url: "index"
- [x] "Templating", url: 'templates'
    - [x] "Templating Basics", url: "handlebars-basics" https://github.com/ember-learn/guides-source/pull/598
    - [x] "Conditionals", url: "conditionals" https://github.com/ember-learn/guides-source/pull/598
    - [x] "Displaying a List of Items", url: "displaying-a-list-of-items" https://github.com/ember-learn/guides-source/pull/598
    - [x] "Displaying the Keys in an Object", url: "displaying-the-keys-in-an-object" https://github.com/ember-learn/guides-source/pull/598
    - [x] "Binding Element Attributes", url: "binding-element-attributes" https://github.com/ember-learn/guides-source/pull/598
    - [x] "Actions", url: "actions" https://github.com/ember-learn/guides-source/pull/598
    - [x] "Built-in Helpers", url: "built-in-helpers" https://github.com/ember-learn/guides-source/pull/598
    - [x] "Writing Helpers", url: "writing-helpers" #598 and #597 
- [x] "Working with JavaScript", url: 'working-with-javascript'
    - [x] "Learning JavaScript", url: "learning-javascript"
    - [x] "Native Classes", url: "native-classes" #596
    - [x] "Classic Classes", url: "classic-classes" #596
    - [x] "Native vs. Classic Class Cheatsheet", url: "native-vs-classic-class-cheatsheet" #596
- [x] "Components", url: 'components'
    - [x] "Component Basics", url: "component-basics"
    - [x] "Defining a Component", url: "defining-a-component" #602
    - [x] "Arguments and Attributes", url: "arguments-and-attributes" #604 
    - [x] "Actions", url: "actions"
    - [x] "Yields", url: "yields"
    - [x] "Interacting with the DOM", url: "interacting-with-the-dom"
    - [x] "Contextual Components", url: "contextual-components"
    - [x] "Classic Components", url: "classsic-components"
- [x] "Routing", url: 'routing'
    - [x] "Introduction", url: "index"
    - [x] "Defining Your Routes", url: "defining-your-routes"
    - [x] "Specifying a Route's Model", url: "specifying-a-routes-model"
    - [x] "Rendering a Template", url: "rendering-a-template"
    - [x] "Redirecting", url: "redirection"
    - [x] "Preventing and Retrying Transitions", url: "preventing-and-retrying-transitions"
      isAdvanced: true
    - [x] "Loading / Error Substates", url: "loading-and-error-substates"
      isAdvanced: true
    - [x] "Query Parameters", url: "query-params"
    - [x] "Asynchronous Routing", url: "asynchronous-routing"
      isAdvanced: true
    - [x] "Controllers", url: 'controllers'
- [ ] "State Management", url: 'state-management'
    - [x] "Overview", url: "index" #595 
    - [x] "Tracked Properties", url: "tracked-properties" #606 
    - [ ] "Patterns for State", url: "patterns-for-state" #580
    - [x] "Computed Properties", url: "computed-properties"
- [x] "Services", url: 'services'
    - [x] Introduction, url: "services" #590 and #590
- 'Data Management', url: 'data-management'
- [ ] "Ember Data", url: 'models' (**skipping this for now**)
    - [x] "Introduction", url: "index" https://github.com/ember-learn/guides-source/pull/629
    - [x] "Defining Models", url: "defining-models"
    - [ ] "Finding Records", url: "finding-records"
    - [ ] "Creating, Updating and Deleting", url: "creating-updating-and-deleting-records"
    - [ ] "Relationships", url: "relationships"
    - [ ] "Pushing Records into the Store", url: "pushing-records-into-the-store"
      isAdvanced: true
    - [ ] "Handling Metadata", url: "handling-metadata"
      isAdvanced: true
    - [x] "Customizing Adapters", url: "customizing-adapters" #578 and #577
      isAdvanced: true
    - [ ] "Customizing Serializers", url: "customizing-serializers"
      isAdvanced: true
- [ ] "Addons and Dependencies", url: "addons-and-dependencies"
    - [ ] "Managing Dependencies", url: "managing-dependencies"
- [x] "Testing", url: 'testing'
    - [x] "Introduction", url: "index" #594
    - [x] "Application Tests", url: "acceptance" #594
    - [x] "Testing Basics", url: "unit-testing-basics" #594
    - [x] "Testing Components", url: "testing-components" #594
    - [x] "Testing Helpers", url: "testing-helpers" #594
    - [x] "Testing Controllers", url: "testing-controllers" #594
    - [x] "Testing Routes", url: "testing-routes" #594
    - [x] "Testing Models", url: "testing-models" #594
- [ ] "Configuration", url: 'configuring-ember'
    - [x] "Configuring Your App", url: "configuring-your-app" #665
    - [x] "Configuring Ember CLI", url: "configuring-ember-cli" #665
    - [x] "Handling Deprecations", url: "handling-deprecations" #665
    - [ ] "Disabling Prototype Extensions", url: "disabling-prototype-extensions"
    - [ ] "Specifying the URL Type", url: "specifying-url-type"
    - [ ] "Embedding Applications", url: "embedding-applications"
    - [ ] "Feature Flags", url: "feature-flags"
      isAdvanced: true
    - [ ] "Optional Features", url: "optional-features"
      isAdvanced: true
    - [ ] "Build targets", url: "build-targets"
    - [ ] "Debugging", url: "debugging"
- [ ] "Deploying", url: 'deploying'
    - [ ] "Overview", url: "index"
- [x] "Upgrading", url: 'upgrading'
    - [x] "Overview", url: 'index'
    - [x] "Editions", url: 'editions' #593 and #580
    - [x] "Deprecations", url: 'deprecations'
- [ ] "Developer Tools", url: "ember-inspector"
    - [ ] "Introduction", url: "index"
    - [ ] "Installing the Inspector", url: "installation"
    - [ ] "Object Inspector", url: "object-inspector"
    - [ ] "The Component Tree", url: "component-tree"
    - [ ] "The View Tree", url: "view-tree"
    - [ ] "Inspecting Routes", url: "routes"
    - [ ] "Data Tab", url: "data"
    - [ ] "Tackling Deprecations", url: "deprecations"
    - [ ] "Library Info", url: "info"
    - [ ] "Debugging Promises", url: "promises"
    - [ ] "Inspecting Objects via the Container", url: "container"
    - [ ] "Rendering Performance", url: "render-performance"
    - [ ] "Troubleshooting", url: "troubleshooting"
- [ ] "Contributing to Ember.js", url: 'contributing' #623 
  isAdvanced: true
    - [ ] "Adding New Features", url: "adding-new-features"
    - [ ] "Repositories", url: "repositories"
- [ ] "Glossary", url: "glossary"
    - [ ] "Web Development", url: "web-development"
- 'Reference', url: 'reference'
    - [ ] "Syntax Conversion Guide", url: "syntax-conversion-guide"
    - [ ] "Accessibility Guide", url: "accessibility-guide"

![the-beacons-are-lit](https://user-images.githubusercontent.com/16627268/54253689-ade43a00-4526-11e9-9e6d-080e135086a9.jpg)

**Comment by muziejus** [comment]
I can do getting started and services, if that's ok?

**Comment by jenweber** [comment]
Thank you! I updated the list with your work!

**Comment by muziejus** [comment]
I'll do the testing section next.

**Comment by betocantu93** [comment]
I can do components section

**Comment by mixonic** [comment]
I'm going through "templating" right now. We will see if I am on track!

**Comment by dennismende** [comment]
I will do Data Management and Ember Data till Friday.

**Comment by richard-viney** [comment]
I'll review the "Working with JavaScript" section.

**Comment by kgautreaux** [comment]
I have submitted a PR for "tracked-properties" so I can claim that post-hoc.

**Comment by mixonic** [comment]
I'm also doing a read through components this morning, mostly with an eye toward using actions consistently as described in https://github.com/ember-learn/guides-source/pull/598#discussion_r265390517.

tl;dr is:

* On native class glimmer components, ember components, and controllers use the `@action` decorator and the element modifier `{{action this.actionMethod}}` passing the method.
* On classic `.extend(` ember components and controllers use the `actions: {` hash and the `{{action` element modifier `{{action 'actionName'}}` passing the action name. This usage is no different than best practices today.

**Comment by richard-viney** [comment]
@mixonic Re. recommended action syntax, I've been a little uncertain on recommended practice of late, as there appears to be no need to use the `action` helper unless you want to add arguments to the invocation. i.e. you can do `@action onClick() {}` in the component and then just `onclick={{this.onClick}}` in the template with no use of the `action` helper at all. Also if you write `{{action this.onClick}}` in the template then you don't actually need the `@action` decorator at all. The latter form does require the `this.` prefix though.

I wonder if we're effectively double-binding the function by using both the `@action` decorator as well as the `action` helper together with `this.` .. ?

**Comment by mixonic** [comment]
@richard-viney Please read the [comment I linked to above](https://github.com/ember-learn/guides-source/pull/598#discussion_r265390517) for a more complete description of what we're suggesting for best practice and why. If you still have questions please find me in `#st-octane` on Discord.

:pray: I'm happy to discuss this topic but lets try to keep comments for this quest issue focused on the work at hand. :beers:

**Comment by muziejus** [comment]
> "Core Concepts", url: "core-concepts" - needs an updated image

I'll make a new image.

**Comment by jason-maier** [comment]
I can work on "Contributing to Ember.js"

**Comment by dennismende** [comment]
@jenweber I took a look at 
- models/defining-models
- models/finding-records
- models/creating-updating-and-deleting-records
and these are all good from my side. @efx already adjusted everything.

**Comment by dennismende** [comment]
@jenweber Could you or someone else provide some guidance in regards to data-management? What is expected here?

**Comment by dennismende** [comment]
@jenweber I will start now with the Routing parts

**Comment by Frozenfire92** [comment]
I will update the image for `"Core Concepts", url: "core-concepts" - needs an updated image`

**Comment by Frozenfire92** [comment]
I will go through Component Basics now

**Comment by balinterdi** [comment]
I'd like to do the following from the ED section:

* Defining Models", url: "defining-models"
* "Finding Records", url: "finding-records"

**Comment by balinterdi** [comment]
I just realized these were done by @efx already (as mentioned by @dennismende above). Shouldn't they be checked out?

**Comment by mike-north** [comment]
I'm going through `computed-properties` now

**Comment by lisaychuang** [comment]
I'll be adding comments to [issue 237](https://github.com/linkedin/css-blocks/issues/237) as I go through the guides, so far I've gone through:

 **"Getting Started", url: 'getting-started'**
- [x] "Quick Start", url: "quick-start"

**Comment by jenweber** [comment]
Hey team, thanks so much for all your help! We are aiming to freeze the guides tonight, and will pick up work again after the keynote. Please make sure to rebase whatever you are working on, as some big sweeping updates just went in for spellchecks, code block accuracy, etc. I will review and hopefully merge any PRs that are open right now before the end of day.

Y'all are the best and I can't wait to show this off at EmberConf this week ❤️

**Comment by jenweber** [comment]
@dennismende to answer your questions:
- We are skipping data management for now
- It is unclear whether link-to will be available in angle brackets for the preview, so let's leave as curly braces for now

We got quite a few updates in for the Routing section today - let me know if you have any to add! Thank you!

**Comment by mixonic** [comment]
Ah, I see we won't land anything tomorrow. I'm going to take on "Configuration", and if I can get that tucked away I'll take another pass at the most important bits even if they already got :eyes:

**Comment by jenweber** [comment]
I'm going to keep working to merge existing PRs tomorrow morning, so for those of you who already got your work PR'd, it will make it in :) 

If others like @mixonic still have time to jump in & help, reviewing other people's PRs and doing code suggestions for typos is helpful! I'm going to sleep now so I will catch y'all tomorrow. This is all super good and we are in excellent shape!

**Comment by jenweber** [comment]
Thanks everyone for your help! As of EmberConf 2019, we reached MVP for the Octane Guides 🎉  

Anyone who wants to see the outcomes can visit https://emberjs.com/editions/octane for the preview guides. Octane isn't yet in a stable release of Ember - this is a preview - but we are in really good shape for when it's time to ship! And now, more people can try it out and share their feedback. Awesome work, everyone!

**Comment by NotGrm** [comment]
I just saw a duplicated paragraph in tutorial, here is the PR to fix it https://github.com/ember-learn/guides-source/pull/1108
